commands/stats.ts

```diff
@@ -0,0 +1,45 @@
+import { Message } from "discord.js";
+import { Command, CommandDefinition, Action } from ".";
+
+export const description: CommandDefinition = {
+  name: "Stat Hoarder",
+  description: "Displays server statistics",
+  usage: ["!stats"],
+  keys: ["stats"],
+};
+
+export const action = (message : Message, key : string) => {
+    const totalMembers= message.guild?.members.cache.size;  
+    let botCount = message.guild?.members.cache.filter(m => m.user.bot).size; 
+    if (botCount !== undefined) botCount++; // Bot doesn't include itself in the botCount lol
+
+    //Get role counts
+    let rollCounts : {rName:string; count:number}[] = [];
+    const rolesJSON : any = message.guild?.roles.cache.toJSON(); // convert role Map Collection to JSON
+    for (const key in rolesJSON) { // Iterate through each role object
+      const roleData : any = rolesJSON[key];
+      const roleName = roleData.name; 
+      const roleCount = message.guild?.roles.cache.get(roleData.id)?.members.size || 0; // Is 0 if undefined
+
+      // If the role is deleted, is @everyone, is a bot role, or has a count of 0, skip
+      if (roleData.deleted || roleName === '@everyone' || roleName === 'YAGPDB.xyz' || roleCount === 0) 
+        continue
+
+      rollCounts.push({
+        rName: roleName,
+        count: roleCount
+      }); 
+    }
+    
+    let data = `Total Members: ${totalMembers}\nBot Count: ${botCount}\n`;
+    for (var index in rollCounts) // Append all rollCounts to the final message to send
+      data += `${rollCounts[index].rName} Members: ${rollCounts[index].count}\n`
+
+    message.author.send(data);
```

Comment:
Do you think it would be better to send this as a `message.reply(str)` ? Might be nice to have this data appear in the chat where it was requested.

Reply:
Yeah absolutely. Since I am not sure how this command will behave with a huge amount of users and roles, I figured to send it to the author first to avoid any clutter. Is there a way we could test this on the SITE server privately?